modules/ref_fb_module/src/classifier_fb_impl.cpp

```diff
@@ -0,0 +1,529 @@
+#include <ref_fb_module/classifier_fb_impl.h>
+#include <ref_fb_module/dispatch.h>
+#include <opendaq/input_port_factory.h>
+#include <opendaq/data_descriptor_ptr.h>
+#include <opendaq/event_packet_ptr.h>
+#include <opendaq/signal_factory.h>
+#include <opendaq/custom_log.h>
+#include <opendaq/event_packet_params.h>
+#include <coreobjects/unit_factory.h>
+#include <opendaq/data_packet.h>
+#include <opendaq/data_packet_ptr.h>
+#include <opendaq/event_packet_ids.h>
+#include <opendaq/packet_factory.h>
+#include <opendaq/range_factory.h>
+#include <opendaq/sample_type_traits.h>
+#include <coreobjects/eval_value_factory.h>
+#include <opendaq/dimension_factory.h>
+
+BEGIN_NAMESPACE_REF_FB_MODULE
+
+namespace Classifier
+{
+
+ClassifierFbImpl::ClassifierFbImpl(const ContextPtr& ctx, const ComponentPtr& parent, const StringPtr& localId)
+    : FunctionBlock(CreateType(), ctx, parent, localId)
+{
+    createInputPorts();
+    createSignals();
+    initProperties();
+}
+
+void ClassifierFbImpl::initProperties()
+{
+    objPtr.addProperty(BoolPropertyBuilder("UseExplicitDimension", false)
+        .setDescription("Choose classification based on input signal min max with step of ClassCount or custom marks").build());
+    objPtr.getOnPropertyValueWrite("UseExplicitDimension") +=
+        [this](PropertyObjectPtr& obj, PropertyValueEventArgsPtr& args) { propertyChanged(true); };
+
+    auto explcitDimensionProp = ListPropertyBuilder("ExplicitDimension", List<Float>()).setVisible(EvalValue("$UseExplicitDimension"))
+        .setDescription("Set custom list for classification rule").build();
+    objPtr.addProperty(explcitDimensionProp);
+    objPtr.getOnPropertyValueWrite("ExplicitDimension") +=
+        [this](PropertyObjectPtr& obj, PropertyValueEventArgsPtr& args) { propertyChanged(true); };
```

Comment:
The final touch here would be to rename the properties to "UseCustomClasses", and "CustomClassList"